src/PollySandbox/ResilienceStrategyStore.cs

```diff
@@ -0,0 +1,288 @@
+﻿// Copyright (c) Martin Costello, 2023. All rights reserved.
+// Licensed under the Apache 2.0 license. See the LICENSE file in the project root for full license information.
+
+using System.Collections.Concurrent;
+using System.Globalization;
+using System.Net;
+using System.Threading.RateLimiting;
+using Microsoft.Extensions.Caching.Memory;
+using Polly;
+using Polly.CircuitBreaker;
+using Polly.RateLimiting;
+using Polly.Retry;
+using Polly.Timeout;
+using Refit;
+
+namespace PollySandbox;
+
+public partial class ResilienceStrategyStore
+{
+    private readonly IMemoryCache _cache;
+    private readonly ILogger _logger;
+    private readonly ConcurrentDictionary<string, ResilienceStrategy> _registry;
+    private readonly IMetricsPublisher _metricsPublisher;
+    private readonly ConcurrentDictionary<string, CircuitBreakerManualControl> _isolators;
+
+    public ResilienceStrategyStore(
+        IMemoryCache cache,
+        IMetricsPublisher metricsPublisher,
+        ILogger<ResilienceStrategyStore> logger)
+    {
+        _cache = cache;
+        _metricsPublisher = metricsPublisher;
+        _logger = logger;
+        _isolators = new();
+        _registry = new();
+    }
+
+    public ResilienceStrategy GetStrategy(string token, ApiEndpointOption endpoint, string resource)
+    {
+        var builder = new ResilienceStrategyBuilder();
```

Comment:
`ResilienceStrategyBuilder` is registered as a transient service in DI. It's better that you resolve it from there. 

That way, it will be correctly configured with telemetry after creation.